notifications_utils/statsd_decorators.py

```diff
@@ -32,3 +33,46 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return time_function
+
+
+def statsd_catch(namespace: str, counter_name: str, exception: Type[Exception]):
+    """Increases a statsd counter when a given exception is raised.
+
+    When the expected exception is raised, the statsd counter will be
+    incremented and the initial exception re-raised again.
+
+    When a non-expected exception is raised, no counter increment
+    occurs and nothing is caught: it should go through with no problem.
+
+    All parameters are required.
+
+    Parameters
+    ----------
+    namespace : str, required
+        The statsd counter namespace.
+
+    counter_name : str, required
+        The statsd counter name.
+
+    exception : Type[Exception]
+        The exception to catch and raise the counter upon.
+
+    Raises
+    ------
+    BaseException
+        Any parameter that is thrown by the decorated method.
+    """
+    def catch_function(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except exception as e:
+                current_app.statsd_client.incr('{namespace}.{counter_name}'.format(
+                    namespace=namespace, counter_name=counter_name
+                ))
```

Comment:
```suggestion
                current_app.statsd_client.incr(f'{namespace}.{counter_name}')
```

Reply:
One of these nice Python 3 feature I need to use as a habit...